### PR TITLE
fix: Added export Details with country code in e-Invoice

### PR DESCRIPTION
**Pull request by DaizyModi**
Partial fix for #732

### CODE REVIEW

File: india_compliance/gst_india/utils/test_e_invoice.py [hunk 1]
```diff
@@ -74,6 +74,41 @@ def test_request_data_for_different_shipping_dispatch_address(self):
...
+        si.update(
+            {
+                "gst_category": "Overseas",
+                "shipping_bill_number": "1234",
+                "shipping_bill_date": frappe.utils.today(),
+                "port_code": "INABG1",
+            }
+        )
+
+        if si.customer_address != si.port_address:
```
Comment (vorasmit) [inline india_compliance/gst_india/utils/test_e_invoice.py]: You should have complete control over the sales invoice that you create and should know if they are equal or not. Better --> Set a port address in the sales invoice.

File: india_compliance/gst_india/utils/e_invoice.py [hunk 2]
```diff
@@ -654,3 +657,30 @@ def get_item_data(self, item_details):
                 "ExpDt": item_details.batch_expiry_date,
             },
         }
+
+    def get_export_details(self):
+        country_code = (
```
Comment (vorasmit) [inline india_compliance/gst_india/utils/e_invoice.py]: Directly from Billing Address only.
Throw if empty. (Add a link to country doctype and e-Invoice Master list)
Throw if the length is not equal to 2. (Add a link to country doctype and e-Invoice Master list)

File: india_compliance/gst_india/utils/transaction_data.py [hunk 3]
```diff
@@ -409,6 +409,9 @@ def get_address_details(self, address_name, validate_gstin=False):
                     **error_context,
                 ),
                 "pincode": int(address.pincode),
+                "country_code": self.sanitize_value(
+                    get_country_code(address.country), min_length=2, max_length=2
```
Comment (vorasmit) [inline india_compliance/gst_india/utils/transaction_data.py]: Set only if not India. No need to sanitize.

File: india_compliance/gst_india/client_scripts/sales_invoice.js [hunk 4]
```diff
@@ -26,5 +26,10 @@ frappe.ui.form.on(DOCTYPE, {
 
     before_submit(frm) {
         frm.doc._submitted_from_ui = 1;
-    }
+    },
+
+    refresh(frm) {
+        frm.set_df_property("port_code", "ignore_validation", 1);
```
Comment (vorasmit) [inline india_compliance/gst_india/client_scripts/sales_invoice.js]: Did you try onload?
Also, `set_df_property` can be part of set_port_code_options

File: india_compliance/public/js/utils.js [hunk 5]
```diff
@@ -84,6 +86,21 @@ Object.assign(india_compliance, {
         if (UNBODY_REGEX.test(gstin)) return "UIN Holders";
         if (OVERSEAS_REGEX.test(gstin)) return "Overseas";
     },
+
+    set_port_code_options(frm) {
```
Comment (vorasmit) [inline india_compliance/public/js/utils.js]: Add these to transaction.js
Utils should ideally not take `frm` as input.

File: india_compliance/gst_india/utils/test_e_invoice.py (hunk 1 above)
Comment (DaizyModi) [reply india_compliance/gst_india/utils/test_e_invoice.py]: Already setting port address from test e_invoice records.

File: india_compliance/gst_india/utils/transaction_data.py (hunk 3 above)
Comment (DaizyModi) [reply india_compliance/gst_india/utils/transaction_data.py]: Used `sanitize_value` for min and max length validations.

**Pull request merged**